stories/wp-dashboard.stories.js

```diff
@@ -59,25 +72,38 @@ storiesOf( 'WordPress', module )
 				requiredScopes: [],
 				grantedScopes: [],
 			} );
-		};
+			dispatch( CORE_MODULES ).receiveGetModules( [
+				{
+					slug: 'analytics',
+					active: true,
+					connected: true,
+				},
+				{
+					slug: 'search-console',
+					active: true,
+					connected: true,
+				},
+			] );
+			dispatch( CORE_USER ).setReferenceDate( '2021-01-23' );
+
+			// For <WPDashboardUniqueVisitors />
+			dispatch( MODULES_ANALYTICS ).receiveGetReport( wpDashboardUniqueVisitorsData, { options: wpDashboardUniqueVisitorsArgs } );
+			dispatch( MODULES_ANALYTICS ).finishResolution( 'getReport', [ wpDashboardUniqueVisitorsArgs ] );
 
-		const addWPSearchConsoleDashboardWidget = createAddToFilter( <WPSearchConsoleDashboardWidget /> );
-		const addWPAnalyticsDashboardWidgetOverview = createAddToFilter( <WPAnalyticsDashboardWidgetOverview /> );
-		const addWPAnalyticsDashboardWidgetTopPagesTable = createAddToFilter( <WPAnalyticsDashboardWidgetTopPagesTable /> );
+			// For <WPDashboardSessionDuration />
+			dispatch( MODULES_ANALYTICS ).receiveGetReport( wpDashboardSessionDurationData, { options: wpDashboardSessionDurationArgs } );
+			dispatch( MODULES_ANALYTICS ).finishResolution( 'getReport', [ wpDashboardSessionDurationArgs ] );
 
-		removeAllFilters( 'googlesitekit.WPDashboardHeader' );
-		removeAllFilters( 'googlesitekit.WPDashboardModule' );
+			// For <WPDashboardImpressions />
+			dispatch( MODULES_SEARCH_CONSOLE ).receiveGetReport( wpDashboardImpressionsData, { options: wpDashboardImpressionsArgs } );
 
-		addFilter( 'googlesitekit.WPDashboardHeader',
-			'googlesitekit.SearchConsole',
-			addWPSearchConsoleDashboardWidget, 11 );
+			// For <WPDashboardClicks />
+			dispatch( MODULES_SEARCH_CONSOLE ).receiveGetReport( wpDashboardClicksData, { options: wpDashboardClicksArgs } );
 
-		addFilter( 'googlesitekit.WPDashboardHeader',
-			'googlesitekit.Analytics',
-			addWPAnalyticsDashboardWidgetOverview, 1 );
-		addFilter( 'googlesitekit.WPDashboardModule',
-			'googlesitekit.Analytics',
-			addWPAnalyticsDashboardWidgetTopPagesTable );
+			// For <WPDashboardPopularPages />
+			dispatch( MODULES_ANALYTICS ).receiveGetReport( wpDashboardPopularPagesData, { options: wpDashboardPopularPagesArgs } );
+			dispatch( MODULES_ANALYTICS ).finishResolution( 'getReport', [ wpDashboardPopularPagesArgs ] );
+		};
```

Comment:
There are just a few more things to clean up here:
* The `doAction` call below was only relevant for the legacy data API which is now no longer used in the WP dashboard widget.
* The `_googlesitekitLegacyData` assignments above are also no longer relevant since they were only used by the previous implementation using the legacy data API.